### ISSUE TITLE
Reported: Regular Expression Denial of Service (ReDoS)

### ISSUE THREAD

**Issue by WilliamHolmes** [ramda/ramda#3195]
https://snyk.io/test/npm/ramda/0.27.1
![image](https://user-images.githubusercontent.com/5538260/139411092-15fcc48a-d090-4c1f-879b-aee407d5bcc0.png)

**Comment by customcommander** [comment]
Not sure but I _think_ this has been fixed in https://github.com/ramda/ramda/pull/3177 so just need a release. (Easier said than done I know :)

**Comment by kz4151** [comment]
Hi. Any updates on when there will be a release created for this vulnerability?

**Comment by CrossEye** [comment]
This is patched.  We are trying to fix one other reported vulnerability (#3192) and release them together, but that one has been a frustrating problem.

**Comment by michele-pinto-kensu** [comment]
> This is patched. We are trying to fix one other reported vulnerability (#3192) and release them together, but that one has been a frustrating problem.

Hi, the other vulnerability is still stuck. Is it possible to release at least this one? Thanks.

**Comment by kz4151** [comment]
Hi @CrossEye. Is there any way that there can be a release created for this vulnerability first?

**Comment by customcommander** [comment]
Unfortunately I think the pull request that intended to fix the issue introduced a failed unit test which is breaking the build. I may have some bandwidth in the coming days to attempt to fix this.

```
  1) trim
       trims all ES5 whitespace:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+ actual - expected

+ '"᠎             　  ﻿Hello, World!\\t\\n\\v\\f\\r   ᠎"'
- '"Hello, World!"'
      + expected - actual

      -"᠎             　  ﻿Hello, World!\t\n\v\f\r   ᠎"
      +"Hello, World!"
      
      at strictEqual (test/shared/eq.js:8:10)
      at Context.eq (test/trim.js:13:5)
      at processImmediate (internal/timers.js:464:21)
```

**Comment by semmel** [comment]
@customcommander 
Removing `\u180E` from the variable `test` in line 6 of `test/trim.js` will probably fix it. `\u180E` is the "Mongolian Vowel Separator" and not recognised as whitespace by "others", meaning at least all JS engine vendors.

I wonder, how could #3177 even got merged with that failing test?

**Comment by adispring** [comment]
I think it should have a plan to release a new version of  ramda, the publish was put off again and again. It was almost two years that ramda did not have a new publish, I don't know why. 

Ramda is popular package, Thousands of  people and project use it, the community's voice should not be ignored again and again.

**Comment by adispring** [comment]
> @customcommander Removing `\u180E` from the variable `test` in line 6 of `test/trim.js` will probably fix it. `\u180E` is the "Mongolian Vowel Separator" and not recognised as whitespace by "others", meaning at least all JS engine vendors.
> 
> I wonder, how could #3177 even got merged with that failing test?

@semmel I found that ramda repository CI/CD not working recently, and I have tried to fix it in another pr: https://github.com/ramda/ramda/pull/3205

**Comment by kz4151** [comment]
Hi. Anymore updates on this issue? Thanks

**Comment by customcommander** [comment]
I am having a look at fixing the build 🤞

**Comment by customcommander** [comment]
See #3206

**Comment by michele-pinto-kensu** [comment]
When are you going to publish a new version of Ramda? thanks

**Comment by customcommander** [comment]
@michele-pinto-kensu Unfortunately with the end of year celebrations coming soon I don't see how we could reasonably manage to get 0.28.0 safely before next year. The last thing we want is more production issues and most of the team away and unable to help.

I took some time recently to get the build pipeline back in order. Now that that is done (yay) I want to make all the preparations so that I can either make that release or assist the person making the release next year.

I make no promise here but I think I speak for most team members when I say that we all want to get 0.28.0 as soon as possible.

**Comment by kz4151** [comment]
Hi @customcommander. Do you have an ETA on when a new release will be out there? Thanks!

**Comment by customcommander** [comment]
@kz4151 I'm currently coordinating with the maintainers to get that done ASAP but I guess not everybody has returned from holidays yet.

**Comment by customcommander** [comment]
See https://github.com/ramda/ramda/issues/3211

**Comment by customcommander** [comment]
Released in `v0.27.2`. See #3216.
